diff --git a/Makefile.am b/Makefile.am
new file mode 100644
index 0000000..6d2bb21
--- /dev/null
+++ b/Makefile.am
@@ -0,0 +1,7 @@
+noinst_LTLIBRARIES = libsha-2.la
+
+libsha_2_la_SOURCES = \
+	sha-256.c \
+	sha-256.h
+
+EXTRA_DIST = LICENSE README.md test.c
diff --git a/configure.ac b/configure.ac
new file mode 100644
index 0000000..a012440
--- /dev/null
+++ b/configure.ac
@@ -0,0 +1,32 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([sha-2], [1.0], [x@y])
+AC_CONFIG_AUX_DIR([auxdir])
+AM_INIT_AUTOMAKE([foreign])
+AC_CONFIG_SRCDIR([sha-256.h])
+AC_CONFIG_HEADERS([config.h])
+AC_CONFIG_MACRO_DIRS([m4])
+LT_INIT
+
+# Checks for programs.
+AC_PROG_CC
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdint.h stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_C_INLINE
+AC_TYPE_SIZE_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([memset])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
diff --git a/sha-256.h b/sha-256.h
index 47f06eb..a4fcff5 100644
--- a/sha-256.h
+++ b/sha-256.h
@@ -1 +1,16 @@
+#ifndef SHA_2_SHA_256_H
+#define SHA_2_SHA_256_H
+#include<stdint.h>
+#include<stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* !defined(__cplusplus) */
+
 void calc_sha_256(uint8_t hash[32], const void *input, size_t len);
+
+#ifdef __cplusplus
+}
+#endif /* !defined(__cplusplus) */
+
+#endif /* !defined(SHA_2_SHA_256_H) */
